editor/editor_properties.h

```diff
@@ -31,13 +31,10 @@
 #ifndef EDITOR_PROPERTIES_H
 #define EDITOR_PROPERTIES_H
 
-#include "editor/create_dialog.h"
-#include "editor/editor_file_system.h"
-#include "editor/editor_inspector.h"
+#include "core/math/expression.h"
```

Comment:
Since those two are only used in the `.cpp` file, that's where they should be included. The `.h` should only have the includes needed for the class declaration.

Some of the other includes that you removed are actually relevant here, as they are used directly: `CreateDialog`, `EditorInspector`, `SceneTreeDialog`, `ColorPickerButton`.
It still compiled without those as they are likely included further down the dependency tree due to the remaining `"editor/property_selector.h"`, but it's better to be explicit about what is actually used here (so that if `PropertySelector`'s dependencies change, `EditorProperty*` won't fail building).

`editor/editor_file_system.h` does seem fine to remove though, I don't see it used in this file.